### PR TITLE
Add decoder fuzzer

### PR DESCRIPTION
**Pull request by kroeckx**
This found CVE-2023-0217

This is waiting for #19669 before merge.

### CODE REVIEW

File: fuzz/decoder.c [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#include "fuzzer.h"
+
+static ASN1_PCTX *pctx;
+
+int FuzzerInitialize(int *argc, char ***argv)
+{
+    FuzzerSetRand();
+
+    OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CRYPTO_STRINGS
+       | OPENSSL_INIT_ADD_ALL_CIPHERS | OPENSSL_INIT_ADD_ALL_DIGESTS, NULL);
```
Comment (paulidale) [inline fuzz/decoder.c:23]: This isn't our usual indentation for continued expressions.

**Review by paulidale — APPROVED** [review]
With the style nit fixed (a few instances below as well).

**Comment by kroeckx** [comment]
https://github.com/openssl/fuzz-corpora/pull/11 should preferably be merged before this

**Review by mattcaswell — APPROVED** [review]

**Comment by tmshort** [comment]
This needs a rebase @kroeckx

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you.

**Pull request closed without merging**